### DISCUSSION TITLE
Why does the Prettier check fail on my PRs, despite formatting everything?

### DISCUSSION THREAD

**Discussion by mybearworld** [ScratchAddons/ScratchAddons#5977] [Help] [answered]
I've only made two PRs so far, but on both, the "Prettify code (outside Scratch Addons)" check fails with this error message:
```
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>

Error: Process completed with exit code 128.
```

Now according to https://github.com/ScratchAddons/ScratchAddons/pull/5960#issuecomment-1519098740 I can just not care, but this doesn't happen on other PRs, so there must be something I am doing wrong, right?

I'm pushing the changes using the built-in Git functionality of VSCode, and formatting the code with the .prettierrc.json in the repository. https://github.com/ScratchAddons/ScratchAddons/actions/runs/4844068522/jobs/8632100559?pr=5960 is an example of it failing.

**Comment by Samq64** [comment, marked as answer]
I'm not sure why it's failing if you already formatted it before commiting, it's possibly be because there's no new line at the end of a file but the easiest way of fixing is to enable the prettier action on your fork so it can format your code automatically.
https://github.com/mybearworld/ScratchAddons/actions

**Reply by mybearworld** [reply]
I just did that - the action also errors, but with a different error:
```
remote: Permission to mybearworld/ScratchAddons.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/mybearworld/ScratchAddons/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```
I'm not sure how to give it permissions to do this.

**Reply by mybearworld** [reply]
By googling the error above, I at least managed to get rid of the X - now there's nothing.

**Comment by DNin01** [comment]
You can see [here](https://github.com/ScratchAddons/ScratchAddons/pull/5960/commits/08b175107777c3ef8c7e14ab4458d6cec3aee75f) what the format-code workflow modified, so now your code is formatted. If the Format code with Prettier status check re-ran when the code was formatted, it would pass, but I've noticed that the checks don't run on the format-code commits for whatever reason, but don't worry, all other checks passed and your code is formatted now.

**Reply by mybearworld** [reply]
Okay, so that means it works now after enabling it on the fork.

**Reply by mybearworld** [reply]
The checks run on the format-code commits now!
